progenitor-impl/src/method.rs

```diff
@@ -2224,18 +2337,207 @@ impl Generator {
                     .add_type_with_name(&schema.to_schema(), Some(name))?;
                 OperationParameterType::Type(typ)
             }
+            BodyContentType::FormData(_) => {
+                // remove FormPart properties
+                let mut schema = item_schema.to_schema();
+                if let schemars::schema::Schema::Object(
+                    schemars::schema::SchemaObject {
+                        object: Some(ref mut object),
+                        ..
+                    },
+                ) = schema
+                {
+                    for param in body_params.iter() {
+                        object.properties.remove_entry(&param.api_name);
+                    }
+                };
+                let name = sanitize(
+                    &format!(
+                        "{}-body",
+                        operation.operation_id.as_ref().unwrap(),
+                    ),
+                    Case::Pascal,
+                );
+                let typ =
+                    self.type_space.add_type_with_name(&schema, Some(name))?;
+                OperationParameterType::Type(typ)
+            }
         };
 
-        Ok(Some(OperationParameter {
+        let body_param = OperationParameter {
             name: "body".to_string(),
             api_name: "body".to_string(),
             description: body.description.clone(),
             typ,
             kind: OperationParameterKind::Body(content_type),
-        }))
+        };
+        body_params.push(body_param);
+
+        Ok(body_params)
+    }
+
+    fn get_form_builder(
+        &self,
+        params: &[OperationParameter],
+        form_ident: &proc_macro2::Ident,
+    ) -> Result<Vec<TokenStream>> {
```

Comment:
each item of the response Vec should represent a similar sort of thing. Currently it looks like kind of a mishmash. I think this would be clearer if we could make each item reflect a similar type of thing and document the concept this function is implementing.